packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -216,17 +241,31 @@ exports.sourceNodes = async (
 
     // Download all files (await for each pool to complete to fix concurrency issues)
     const fileNodes = [...nodes.values()].filter(isFileNode)
+
     if (fileNodes.length) {
       const downloadingFilesActivity = reporter.activityTimer(
         `Remote file download`
       )
       downloadingFilesActivity.start()
-      await asyncPool(concurrentFileRequests, fileNodes, async node => {
-        await downloadFile(
-          { node, store, cache, createNode, createNodeId, getCache, reporter },
-          pluginOptions
-        )
-      })
+      try {
+        await asyncPool(concurrentFileRequests, fileNodes, async node => {
+          await downloadFile(
+            {
+              node,
+              store,
+              cache,
+              createNode,
+              createNodeId,
+              getCache,
+              reporter,
+            },
+            pluginOptions
+          )
+        })
+      } catch (e) {
+        gracefullyRethrow(downloadingFilesActivity, e)
+        return
+      }
       downloadingFilesActivity.end()
```

Comment:
Nit:
with finally you don't have to pass the activity around

```suggestion
      } catch (e) {
        gracefullyRethrow(e)
        return
      } finally {
        downloadingFilesActivity.end();
      }
```

Reply:
But this is doing different thing what I'm doing, because `.end()` is not the only thing I'm doing on `activity` if error was thrown - https://github.com/gatsbyjs/gatsby/pull/25447#discussion_r448523491 applies here

I do want to run:
```
  if (activity.panicOnBuild) {
    activity.panicOnBuild(error)
  }
```

in case of error, so using `finally` would just split things apart and I would still need to make use of `activity` in `catch` handler

Reply:
Oh, I missed activity.panicOnBuild, I thought it was using reporter. NVM!